src/main/java/mate/academy/springbootwebgreqit/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package mate.academy.springbootwebgreqit.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootwebgreqit.dto.order.OrderRequestDto;
+import mate.academy.springbootwebgreqit.dto.order.OrderResponseDto;
+import mate.academy.springbootwebgreqit.dto.orderItem.OrderItemResponseDto;
+import mate.academy.springbootwebgreqit.mapper.OrderItemMapper;
+import mate.academy.springbootwebgreqit.mapper.OrderMapper;
+import mate.academy.springbootwebgreqit.model.Order;
+import mate.academy.springbootwebgreqit.model.OrderItem;
+import mate.academy.springbootwebgreqit.model.User;
+import mate.academy.springbootwebgreqit.repository.OrderItemReposirory;
+import mate.academy.springbootwebgreqit.repository.OrderRepository;
+import mate.academy.springbootwebgreqit.service.OrderService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemReposirory orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public OrderResponseDto addOrder(OrderRequestDto requestDto, User user) {
+        Order order = orderMapper.toModel(requestDto);
```

Comment:
First check if shop cart if empty 
```suggestion
        ShoppingCart cart = cartRepository.findByUserId(userId);
        if (cart.getCartItems().isEmpty()) {
```

Reply:
Also clear shop cart after replace order